dolphin/api/v1/pools.py

```diff
@@ -12,19 +12,79 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+import six
+from webob import exc
+
+from dolphin import db, exception
+from dolphin.api import common_utils as common
 from dolphin.api.common import wsgi
+from dolphin.api.views import pools as pool_view
+from dolphin.i18n import _
 
 
 class PoolController(wsgi.Controller):
+    def _verify_storage(self, context, storage_id):
+        """verify the storage association  ."""
+        try:
+            db.storage_get(context, storage_id)
+        except exception.StorageNotFound:
+            msg = _("storage  '%s' not found.") % storage_id
+            raise exc.HTTPNotFound(explanation=msg)
+
+    def _get_pools_search_options(self):
+        """Return pools search options allowed ."""
+        return ('name', 'status', 'id', 'storage_id')
+
+    def _show(self, req, pool_id, storage_id=None):
+        ctxt = req.environ['dolphin.context']
+        if storage_id:
+            self._verify_storage(ctxt, storage_id)
+        try:
+            pool = db.pool_get(ctxt, pool_id)
```

Comment:
Imagine a scenario of valid storage id and valid poolId is passed but they are of 2 different backends. Currently pool_get is not considering storage_id

Reply:
good catch! handled. thanks